pkgs/development/libraries/vtk/default.nix

```diff
@@ -5,24 +5,28 @@ with stdenv.lib;
 
 let
   os = stdenv.lib.optionalString;
-  majorVersion = "5.10";
-  minorVersion = "1";
+  majorVersion = "7.0";
+  minorVersion = "0";
   version = "${majorVersion}.${minorVersion}";
 in
 
 stdenv.mkDerivation rec {
   name = "vtk-${os (qtLib != null) "qvtk-"}${version}";
   src = fetchurl {
-    url = "${meta.homepage}files/release/${majorVersion}/vtk-${version}.tar.gz";
-    sha256 = "1fxxgsa7967gdphkl07lbfr6dcbq9a72z5kynlklxn7hyp0l18pi";
+    url = "${meta.homepage}files/release/${majorVersion}/VTK-${version}.tar.gz";
+    sha256 = "1hrjxkcvs3ap0bdhk90vymz5pgvxmg7q6sz8ab3wsyddbshr1abq";
   };
 
-  # https://bugzilla.redhat.com/show_bug.cgi?id=1138466
-  postPatch = "sed '/^#define GL_GLEXT_LEGACY/d' -i ./Rendering/vtkOpenGL.h";
-
   buildInputs = [ cmake mesa libX11 xproto libXt ]
     ++ optional (qtLib != null) qtLib;
 
+  preBuild = ''
+    LD_LIBRARY_PATH="$(pwd)/lib:$LD_LIBRARY_PATH";
```

Comment:
You will need to `export` this value if you want it to work. As in `export LD_LIBRARY_PATH="$(pwd)/lib"`.
Without exporting the value, it will have no impact, as you surely observed with your `echo`.

The `:$LD_LIBRARY_PATH` is useless here because `LD_LIBRARY_PATH` is empty, so it just ads a colon at the end of $(pwd)/lib.